apilib/model.py

```diff
@@ -39,6 +39,19 @@ def __init__(self, **kwargs):
                 raise exceptions.UnknownFieldException('Unknown field "%s"' % key)
             setattr(self, key, value)
 
+    def validate(self, error_context=None, context=None):
+        is_root = not error_context
+        error_context = error_context or ErrorContext()
+        context = self.make_parent_context(self.to_dict(), context) if context else None
+        for field_name, field in six.iteritems(self._field_name_to_field):
+            field._validate(getattr(self, field_name), error_context.extend(field=field_name), context)
+
+        if error_context.has_errors():
+            if is_root:
```

Comment:
Since during `invoke`, there is no root, when does this situation happen? If you manually called `validate()` on a model outside of the context of a method?

Reply:
Yeah I'm not sure how necessary it is, since an `error_context` is being provided wherever this is called. That said, I found this part a bit confusing--I'm actually not sure where is_root would be true for the `from_json` code I copied this from.